process/smartContract/hooks/blockChainHook.go

```diff
@@ -393,6 +396,76 @@ func (bh *BlockChainHookImpl) LastEpoch() uint32 {
 	return 0
 }
 
+func (bh *BlockChainHookImpl) getCurrentEpochStartBlockHeader() (data.HeaderHandler, error) {
+	currentEpoch := bh.CurrentEpoch()
+
+	if currentEpoch == 0 {
+		return bh.blockChain.GetGenesisHeader(), nil
+	}
+
+	epochStartBlock, _, err := process.GetBlockHeaderForEpochStartFromStorage(
+		currentEpoch,
+		bh.shardCoordinator.SelfId(),
+		bh.storageService,
+		bh.uint64Converter,
+		bh.marshalizer,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return epochStartBlock, nil
+}
+
+// RoundTime returns the duration of a round
+func (bh *BlockChainHookImpl) RoundTime() uint64 {
+	if bh.nodesSetup == nil {
+		return 0
+	}
+
+	return bh.nodesSetup.GetRoundDuration()
+}
+
+// EpochStartBlockTimeStamp returns the timestamp of the first block of the current epoch
+func (bh *BlockChainHookImpl) EpochStartBlockTimeStamp() uint64 {
+	defer stopMeasure(startMeasure("EpochStartBlockTimeStamp"))
+
+	epochStartBlock, err := bh.getCurrentEpochStartBlockHeader()
+
+	if err != nil {
```

Comment:
I think we should propagate the error to the caller instead of giving a default value.

The timestamp value might be used by contracts to compute rewards, and giving back 0 would possibly lead to drained contracts in case of a generalized bug in these cases.